src/sql/workbench/contrib/query/browser/fileQueryResultsWriter.ts

```diff
@@ -0,0 +1,336 @@
+
+				if (m.message.includes('Total execution time')) {
+					this.closingMessageIncluded = true;
+				}
+
+				this.messages.push(m);
+			});
+		}
+		else {
+			this.messages.push(incomingMessage);
+		}
+
+		if (!this.hasCreatedResultsFile && this.closingMessageIncluded && this.queryContainsError && this.numQueriesToFormat === 0) {
+			await this.createResultsFile();
+		}
+	}
+
+	public reset() {
+		this.messages = [];
+		this.formattedQueryResults = [];
+		this.tables = [];
+		this.numQueriesToFormat = 0;
+		this.queryContainsError = false;
+		this.closingMessageIncluded = false;
+		this.hasCreatedResultsFile = false;
+	}
+
+	private addResultSet(resultSetSummaries: ResultSetSummary[]) {
+		const tables: Array<Table<any>> = [];
+
+		for (const resultSetSummary of resultSetSummaries) {
+			// ensure we aren't adding a resultSet that has already been added
+			if (this.tables.find(t => t.resultSet.batchId === resultSetSummary.batchId && t.resultSet.id === resultSetSummary.id)) {
+				continue;
+			}
+
+			const table = this.instantiationService.createInstance(Table, this.runner, resultSetSummary, this);
+			tables.push(table);
+		}
+
+		this.tables = this.tables.concat(tables);
+	}
+
+	public async aggregateQueryResults(results: string) {
+		this.numQueriesToFormat--;
+		this.formattedQueryResults.push(results);
+
+		if (this.numQueriesToFormat === 0) {
+			await this.createResultsFile();
+		}
+	}
+
+	public async createResultsFile() {
+		let fileContent = this.mergeQueryResultsWithMessages();
+
+		let fileData = fileContent.map(m => {
+			if (m.includes('rows affected') || m.includes('row affected')) {
+				return '\r\n' + m + '\r\n';
+			}
+			else if (m.includes('Total execution time:')) {
+				return '\r\n' + m;
+			}
+
+			return m;
+		}).join('\r\n');
+
+		const input = this.untitledEditorService.create({ initialValue: fileData });
+		await input.resolve();
+		input.setDirty(false);
+		await this.instantiationService.invokeFunction(formatDocumentWithSelectedProvider, input.textEditorModel, FormattingMode.Explicit, Progress.None, CancellationToken.None);
+		this.hasCreatedResultsFile = true;
+
+		return this.editorService.openEditor(input);
+	}
+
+	private mergeQueryResultsWithMessages() {
+		let content: Array<string> = [];
+		let extractedMessages = this.messages.map(m => m.message);
+
+		// Merges query result set before respective '# row(s) affected' message
+		while (extractedMessages.length > 0 && this.formattedQueryResults.length > 0) {
+			if (extractedMessages[0]?.includes('rows affected') || extractedMessages[0]?.includes('row affected')) {
+				content.push('\r\n' + this.formattedQueryResults.shift());
+			}
+
+			content.push(extractedMessages.shift());
+		}
+
+		// append any remaining query results and messages.
+		return [...content, ...this.formattedQueryResults, ...extractedMessages];
+	}
+}
+
+export class Table<T> extends Disposable {
+	public isOnlyTable: boolean = true;
+	public gridDataProvider: IGridDataProvider;
```

Comment:
do these fields need to be public.  I didn't see them referenced outside the class.